### ISSUE TITLE
YAML::Node::as<T> returns const T

### ISSUE THREAD

**Issue by peterhuene** [jbeder/yaml-cpp#339]
`YAML::Node::as<T>` (and its overload) is declared as:

```c++
template <typename T> const T as() const;
```

When invoked as:

```c++
string foo = node.as<string>();
```

This invokes the copy constructor for `string` instead of the move constructor because `T const&&` will bind to the former over the latter.  This results in a copy of a perfectly good temporary that could have been moved from.

As far as I can tell the return value does not need to be const.

**Comment by jbeder** [comment]
Good point. Fixed, 97d56c3f3608331baaee26e17d2f116d799a7edc.

**Comment by peterhuene** [comment]
Thanks!
